worker/uniter/runner/jujuc/context.go

```diff
@@ -163,13 +163,18 @@ type ContextUnit interface {
 	CloudSpec() (*params.CloudSpec, error)
 }
 
+type UpsertArgs struct {
+	Value          secrets.SecretValue
```

Comment:
Should these values themselves be optional?

Reply:
Value is an interface so nil supported.
With Rotate, a negative value was going to mean "unset". Easier than using pointers IMO. I'll add a comment